trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/remote/VirtualHostStyleRemoteS3Facade.java

```diff
@@ -28,9 +28,10 @@ public class VirtualHostStyleRemoteS3Facade
     private final boolean https;
     private final Optional<Integer> port;
 
-    public VirtualHostStyleRemoteS3Facade()
+    @Inject
+    public VirtualHostStyleRemoteS3Facade(RemoteS3Config remoteS3Config)
     {
-        this(VIRTUAL_HOST_BUILDER, true, Optional.empty());
+        this((bucket, region) -> RemoteS3HostBuilder.getHostName(bucket, region, remoteS3Config.getDomain(), remoteS3Config.getVirtualHostStyleTemplate()), remoteS3Config.getHttps(), remoteS3Config.getPort());
```

Comment:
As per my other comment (https://github.com/trinodb/aws-proxy/pull/116/files#r1687807514), I do not believe this would work for an operation that has no bucket. The template would render with an empty string for `${bucket}` resulting in an invalid domain.

We can implement some logic to strip leading dots (e.g., for templates like `${bucket}.s3.example.com`) or repeated dots (e.g., `s3.${bucket}.example.com`), or just have 2 templates